spring/boot2-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaHttpHeadersUtil.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.spring.web.reactive;
+
+import java.util.Map.Entry;
+
+import org.springframework.http.HttpHeaders;
+
+import com.linecorp.armeria.common.HttpHeaderNames;
+
+import io.netty.util.AsciiString;
+
+final class ArmeriaHttpHeadersUtil {
+
+    static HttpHeaders fromArmeriaHttpHeaders(com.linecorp.armeria.common.HttpHeaders armeriaHeaders) {
```

Comment:
Can't we use `ArmeriaHttpUtil.toNettyHttp1ClientHeaders()`?

Reply:
We can't because this converts to Spring `HttpHeaders(org.springframework.http.HttpHeaders)` but the method converts to Netty `HttpHeaders(io.netty.handler.codec.http)`. 😄

Reply:
I am concerned that some disallowed HTTP/2 headers unfiltered by this method could be added to `springHeaders`.
How about adding a variant of ArmeriaHttpUtil.toNettyHttp1ClientHeaders() like `ArmeriaHttpUtil.toHttp1ClientHeaders()`?
```java
void toHttp1ClientHeaders(
    HttpHeaders inputHeaders, BiConsumer<CharSequence, String> outputHeadersWriter) {

}

var springHeaders = ...;
toHttp1ClientHeaders(armeriaHeaders, (key, value) -> springHeaders.add(key, value));
```

Reply:
Let me just run a quick benchmark to see if there's any degradation using `BiConsumer`. If there's not, we can gather the same logic in the method:
https://github.com/line/armeria/blob/master/tomcat9/src/main/java/com/linecorp/armeria/server/tomcat/TomcatService.java#L527-L551
https://github.com/line/armeria/blob/master/jetty9/src/main/java/com/linecorp/armeria/server/jetty/JettyService.java#L378-L390

> I am concerned that some disallowed HTTP/2 headers unfiltered by this method could be added to springHeaders.

Could you tell me which one you are concerned?

Reply:
Had a chat with the team and decided to create a method in `ArmeriaHttpUtil` to dedup the logic. 😉

Reply:
> Could you tell me which one you are concerned?

If an HTTP/2 request is downgraded to HTTP/1, we may need to remove disallowed headers from HTTP/2 headers as we did in `toNettyHttp1ClientHeaders`.

Reply:
Got it. 😄
Because this will be also used to convert Armeria HTTP/1.1 headers(,which has pseudo headers,) to other framework headers, let's keep it as it is. 😉